library/core/src/main/java/com/google/android/exoplayer2/text/webvtt/WebvttParserUtil.java

```diff
@@ -74,7 +84,21 @@ public static long parseTimestampUs(String timestamp) throws NumberFormatExcepti
     if (parts.length == 2) {
       value += Long.parseLong(parts[1]);
     }
-    return value * 1000;
+
+    return adjustCueForPtsWrap(value);
+  }
+
+  /**
+   * Convert a WebVTT Cue timestamp to PTS, adjusting for PTS Wrap.  According to the Pantos spec (last paragraph of
+   * <a href="https://tools.ietf.org/html/draft-pantos-hls-rfc8216bis-04#section-3.5">section 3.5 </a>) the
+   * Cue timestamp can be non-monotonic, that is it <b>does not</b> wrap with the PTS.
+   *
+   * @param webvttCueTimestampMs - the timestamp in ms parsed from the VTT cue (non-monotonic)
+   * @return cue time, in ms adjusted for any PTS wrap.
+   */
+  private static long adjustCueForPtsWrap(long webvttCueTimestampMs) {
+    long cueAsPts = (webvttCueTimestampMs * 90) % (TimestampAdjuster.MAX_PTS_PLUS_ONE - 1);
+    return TimestampAdjuster.ptsToUs(cueAsPts);
   }
```

Comment:
All WebVTT timestamps pass through this logic, including the LOCAL time, so that should cover all the possible interpretations of Section 3.5 last paragraph.  Note, if the timestamp is less then 1 day, 2:30:43.717677 (33 bit PTS wrap max value), then this code becomes a noop.  If it's over that, then the actual value is the remainder (line 100).